docs: Move Localnet development to separate page (#977)
oasisprotocol/docs: docs/dapp/tools/localnet.md

> # Local Development

Reviewer: I think we need to have the "Localnet" term in the title. So "Localnet Development", "Using Localnet", "Running Localnet" or just "Localnet".

For convenient development and testing of your dApps the Oasis team prepared
both [ghcr.io/oasisprotocol/sapphire-localnet][sapphire-localnet] and
[ghcr.io/oasisprotocol/emerald-localnet][emerald-localnet] Docker images.
They will bring you a complete Oasis network stack to your desktop. The
Localnet Sapphire instance **mimics confidential transactions**, but it does
not run in a trusted execution environment nor does it require Intel's SGX on
your computer. The network is isolated from the Mainnet or Testnet and consists
of a:

- single Oasis validator node with 1-second block time and 30-second epoch,
- single Oasis client node,

> - single compute node running Oasis Sapphire,

Reviewer 2: ```suggestion
- single compute node running Oasis Sapphire or Emerald,
```

- single key manager node,
- PostgreSQL instance,
- Oasis Web3 gateway with transaction indexer and enabled Oasis RPCs,
- helper script which populates the account(s) for you.

> *Note: You will need at least 16GB of memory.*

Reviewer: Let's use admonition. Also, looking at terminology we use [here](https://docs.oasis.io/node/run-your-node/prerequisites/hardware-recommendations#memory). Something like this?

```suggestion
:::note Hardware requirements

You will need at least 16GB of RAM to spin up the docker image apart from the OS on your machine.

:::
```

To run the image, execute:

```sh

> docker run -it -p8545:8545 -p8546:8546 ghcr.io/oasisprotocol/sapphire-localnet

Reviewer: Since we have Envoy support listening on 8544, this should become:

```suggestion
docker run -it -p8544-8546:8544-8546 ghcr.io/oasisprotocol/sapphire-localnet
```

Also below.

```

> or substitute "sapphire" for "emerald"

Reviewer: What if we used a tab widget with "Sapphire Localnet" and "Emerald Localnet" tabs to separate the shell commands?

```sh
docker run -it -p8545:8545 -p8546:8546 ghcr.io/oasisprotocol/emerald-localnet
```

> After a while, the tool will show you something like this:

Reviewer: ```suggestion
After a while, running the `sapphire-localnet` will show you something like this:
```

(1) 0xa521f94f8a38b1d027D526017EB229327B9D6cA0 (10000 TEST)
(2) 0x1e0f8369215D6C5Af5E14eD6A0D6ae7372776A79 (10000 TEST)
(3) 0xB60cA28B491747a27C057AdBF3E71F3CCC52332C (10000 TEST)
(4) 0x88D7d924e521a6d07008a373D5b33281148ffEDc (10000 TEST)

Private Keys
==================
(0) 0x617346c545d62b8213ea907acf1b570a7405683e2c6dcaf963fc21fd677e0c56
(1) 0xf82d6e09208b0bd44a397f7e73b05c564e6c9f70b151ee7677e2bb8d6ce5d882
(2) 0xeb2f21d20086f3dd6bfe7184dad1cb8b0fb802f27b1334e836a19eda0a43a1c2
(3) 0x82b0203d6063992b1052004b90411c45d4f3afab696346f006e74c6abd8f855e
(4) 0x7179c6e1add3a2993822653b9c98fe606f47fb6d4c0d0d81b31b067cf6bb5f83

HD Wallet
==================
Mnemonic:       coach genre beach child crunch champion tell adult critic peace canoe stable
Base HD Path:   m/44'/60'/0'/0/%d

WARNING: The chain is running in ephemeral mode. State will be lost after restart!

 * Listening on http://localhost:8545 and ws://localhost:8546. Chain ID: 0x5afd
 * Container start-up took 66 seconds, node log level is set to warn.
```

Those familiar with local dApp environments will find the output above similar
to `geth --dev` or `ganache-cli` commands or the `geth-dev-assistant` npm

> package. [sapphire-localnet] will spin up a private Oasis Network locally,

Reviewer 2: ```suggestion
package. The [sapphire-localnet] will spin up a private Oasis Network locally,
```
else there is missing capitalisation?
